corehq/apps/hqwebapp/forms.py

```diff
@@ -21,6 +24,23 @@ def clean_username(self):
         username = self.cleaned_data['username'].lower()
         return username
 
+    def clean(self):
+        lockout_message = mark_safe(_('Sorry - you have attempted to login with an incorrect password too many times. Please <a href="/accounts/password_reset_email/">click here</a> to reset your password.'))
+        username = self.cleaned_data['username']
+        try:
+            cleaned_data = super(EmailAuthenticationForm, self).clean()
+        except ValidationError:
+            user = CouchUser.get_by_username(username)
+            if user and user.is_web_user() and user.is_locked_out():
+                raise ValidationError(lockout_message)
+            else:
+                raise
+        user = CouchUser.get_by_username(username)
+        if user and user.is_web_user() and user.is_locked_out():
+            raise ValidationError(lockout_message)
```

Comment:
Agreed this is super ugly. `get_by_username` can definitely be above the block, and I will make that change. The try statement is where credentials are checked, and therefore is_locked_out() can change as a result of that code block. I used this logic so a 5th login failure returns the lockout message, as that is when lockout actually starts. Otherwise, the 6th attempt is required to get the message but will fail no matter what. It seemed slightly nicer to the user, but if that is not worth the code ugliness I am happy to change it.